lib/money/rates_store/memory.rb

```diff
@@ -31,16 +33,21 @@ def initialize(opts = {}, rates = {})
       # @param [String] currency_iso_from Currency to exchange from.
       # @param [String] currency_iso_to Currency to exchange to.
       # @param [Numeric] rate Rate to use when exchanging currencies.
+      # @param [Date] date Date of the exchange rate, ignored if not using a historical store.
```

Comment:
I wonder if we should allow storing this at any resolution (month, day, hour, etc), which might be important for some applications

Reply:
I suppose this is related to the [discussion about string formatting](https://github.com/RubyMoney/money/pull/962#discussion_r556966661), if the format was chosen by the caller, they would be free to implement storing other resolutions.

But I'm really wondering now if this is another sign of how this is really a problem of the store keying, completely unrelated to the actual storage type (in process memory vs somewhere else) and if maybe I should try splitting this behavior out completely, even if it would be a breaking change.